dashboard/app/models/user.rb

```diff
@@ -157,6 +157,12 @@ class User < ActiveRecord::Base
   accepts_nested_attributes_for :school_info, reject_if: :preprocess_school_info
   validates_presence_of :school_info, unless: :school_info_optional?
 
+  validate :must_be_a_teacher_if_has_sections, on: :update, if: -> {user_type_changed? && student?}
```

Comment:
No, I didn't clear it with anyone. Perhaps trying to prevent the user type change is too much / unnecessary? I can pull that out and just do the section invalidation when a teacher doesn't exist, and the not-found redirect trying to join an invalid section. In each of these cases, we blow up with an error now so I wouldn't be breaking functionality. What do you think?